spec/unit/type/k5login_spec.rb

```diff
@@ -69,6 +77,18 @@ def resource(attrs = {})
           end
         end
 
+        context "with selinux" do
```

Comment:
I started trying to debug the unit tests failures on this, which at very least are failing due to some issues with the mocking. I was able to get the test mostly working by restructuring it like so:
```
        context "with selinux" do
          Puppet::Util::SELinux.stubs(:selinux_support?).returns true
          Puppet::Util::SELinux.stubs(:get_selinux_current_context).returns "user_u:role_r:type_t:s0"
          subject { resource(:content => "daniel@EXAMPLE.COM\n",).retrieve }
          it "should return correct values based on SELinux state" do
            expect(subject[:seluser]).to eq("user_u")
            expect(subject[:selrole]).to eq("role_r")
            expect(subject[:seltype]).to eq("type_t")
            expect(subject[:selrange]).to eq("s0")
          end
        end
```

The issue now is this apparently doesn't correctly stub `selinux_support?`, it's still false. Not sure why.

Reply:
@jhelwig and I figured out what's going on here. It has to do with how Puppet dynamically creates and names classes, and took a bit of digging to figure out what the class name should to be.

This will get the tests green.
```diff
diff --git a/spec/unit/type/k5login_spec.rb b/spec/unit/type/k5login_spec.rb
index a3aeae5..291c6a8 100644
--- a/spec/unit/type/k5login_spec.rb
+++ b/spec/unit/type/k5login_spec.rb
@@ -78,10 +78,12 @@ describe Puppet::Type.type(:k5login), :unless => Puppet.features.microsoft_windo
         end

         context "with selinux" do
-          Puppet::Util::SELinux.stubs(:selinux_support?).returns true
-          Puppet::Util::SELinux.stubs(:get_selinux_current_context).returns "user_u:role_r:type_t:s0"
           subject { resource(:content => "daniel@EXAMPLE.COM\n",).retrieve }
           it "should return correct values based on SELinux state" do
+            Puppet::Type::K5login::ProviderK5login.any_instance.stubs(:selinux_support?).returns true
+            Puppet::Type::K5login::ProviderK5login.any_instance.stubs(:get_selinux_current_context).returns "user_u:role_r:type_t:s0"
             expect(subject[:seluser]).to eq("user_u")
             expect(subject[:selrole]).to eq("role_r")
             expect(subject[:seltype]).to eq("type_t")
```

Reply:
I'll get that change in place and pushed up shortly.